crates/vm/levm/src/vm.rs

```diff
@@ -379,9 +367,11 @@ impl VM {
             .ok_or(VMError::InsufficientAccountFunds)?;
 
         // (2) INSUFFICIENT_ACCOUNT_FUNDS
-        if sender_account.info.balance < up_front_cost {
-            return Err(VMError::InsufficientAccountFunds);
-        }
+        sender_account.info.balance = sender_account
+            .info
+            .balance
+            .checked_sub(up_front_cost)
+            .ok_or(VMError::InsufficientAccountFunds)?;
```

Comment:
I don't know the difference between this check and the (5), here is assigned when it should not (because previously did not perform the operation).

Reply:
(we talked about this afterwards)